routing/endpointregistry.go

```diff
@@ -115,23 +159,72 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	return routes
 }
 
+func (r *EndpointRegistry) updateStats() {
+	if !r.passiveHealthCheckEnabled {
+		return
+	}
+	ticker := time.NewTicker(r.statsResetPeriod)
+
+	for {
+		r.data.Range(func(key, value any) bool {
+			e := value.(*entry)
+
+			curSlot := e.curSlot.Load()
+			nextSlot := (curSlot + 1) % 2
+			e.totalFailedRoundTrips[nextSlot].Store(0)
+			e.totalRequests[nextSlot].Store(0)
+
+			failed := e.totalFailedRoundTrips[curSlot].Load()
+			requests := e.totalRequests[curSlot].Load()
+			failedRoundTripsRatio := float64(failed) / float64(requests)
+			if failedRoundTripsRatio > r.minFailed && requests > r.minRequests {
```

Comment:
Lets check whether we need `failedRoundTripsRatio > r.minFailed`

Reply:
I still would like to keep it, because I can imagine some healthy node to fail only single request because of some glitch. Therefore failedRoundTripsRatio would be equal to something small, like 0.005. Giving such small values, I would like to ignore such values at all, and that's why I introduce `minFailed` parameter (which I will rename to `minFailedRatio`.

Reply:
@RomanZavodskikh the probability would be so low that we would not have a lot of traffic impaired by that and other pods will be able to take the load.